### PR TITLE
FI-1581: Check that output file array contains absolute urls 

### PR DESCRIPTION
**Pull request by Jack-Fraser16**
Implementing [this ticket.](https://oncprojectracking.healthit.gov/support/secure/RapidBoard.jspa?rapidView=40&projectKey=FI&view=detail&selectedIssue=FI-1581&quickFilter=183)

### CODE REVIEW

File: lib/onc_certification_g10_test_kit/bulk_data_group_export_stu1.rb [hunk 1]
```diff
@@ -244,7 +244,8 @@ class BulkDataGroupExportSTU1 < Inferno::TestGroup
 
         Each file SHALL contain resources of only one type, but a server MAY create more than one file for each resource type returned.
 
-        * url - the path to the file. The format of the file SHOULD reflect that requested in the _outputFormat parameter of the initial kick-off request.
+        * url - the #{'absolute' if config.options[:require_absolute_urls_in_output]} path to the file.
```
Comment (Jack-Fraser16) [inline lib/onc_certification_g10_test_kit/bulk_data_group_export_stu1.rb:247]:
```suggestion
        * url - the #{'absolute' if config.options[:require_absolute_urls_in_output]} path to the file.
```
This string interpolation isn't working as intended -- seems _config.options_ is an empty hash at the time of interpolation. If there's a quick alternative, let me know.

Comment (Jammjammjamm) [reply lib/onc_certification_g10_test_kit/bulk_data_group_export_stu1.rb:247]: Descriptions are evaluated at boot time, and outside of the context of a test session, so suite options are unavailable. Hacking in a workaround doesn't seem worth it for such a minor change.

File: lib/onc_certification_g10_test_kit/bulk_data_group_export_stu1.rb [hunk 2]
```diff
@@ -265,6 +265,10 @@ class BulkDataGroupExportSTU1 < Inferno::TestGroup
           ['type', 'url'].each do |key|
             assert file.key?(key), "Output file did not contain \"#{key}\" as required"
           end
+
+          if config.options[:require_absolute_urls_in_output]
+            assert file['url'].match?(%r{\Ahttps?://}), 'URLs in output file were not absolute as required.'
```
Comment (Jammjammjamm) [inline lib/onc_certification_g10_test_kit/bulk_data_group_export_stu1.rb:270]: I think this error message should be something like:
```
"URLs in output file must be absolute, but found `#{file['url']}`"
```
Also, we've verified that the `url` key is present, but not that its value is a String, so it isn't safe to call `match` on it.

Comment (Jack-Fraser16) [reply lib/onc_certification_g10_test_kit/bulk_data_group_export_stu1.rb]: Added `to_s` which is defined on each data type, so should be fine?

**Review by Jammjammjamm — APPROVED** [review]

**Comment by arscan** [comment]
I'm going to go ahead and merge @Jack-Fraser16 because it is approved.

**Pull request merged**